api/serializers.py

```diff
@@ -33,3 +33,54 @@ def validate(self, data):
         except IOC.DoesNotExist:
             data["found"] = False
         return data
+
+
+class FeedSerializer(serializers.Serializer):
+    feed_type = serializers.CharField(max_length=20, write_only=True)
+    attack_type = serializers.CharField(max_length=20, write_only=True)
+    age = serializers.CharField(max_length=10, write_only=True)
+    format_ = serializers.CharField(max_length=5, write_only=True)
+    value = serializers.CharField(read_only=True, source="name")
+    scanner = serializers.BooleanField(read_only=True)
+    payload_request = serializers.BooleanField(read_only=True)
+    first_seen = serializers.DateTimeField(read_only=True)
+    last_seen = serializers.DateTimeField(read_only=True)
+    times_seen = serializers.IntegerField(read_only=True)
+
+    def validate(self, data):
+        """
+        Check a given value against a list of valid values
+        """
+        # data is a dictionary
+        feed_type = data["feed_type"]
+        attack_type = data["attack_type"]
+        age = data["age"]
+        format_ = data["format_"]
+
+        # valid values
+        feed_choices = ["log4j", "cowrie", "all"] + [
+            x.lower() for x in GENERAL_HONEYPOTS
+        ]
+        attack_types = ["scanner", "payload_request", "all"]
+        age_choices = ["persistent", "recent"]
+        formats = ["csv", "json", "txt"]
+
+        if feed_type not in feed_choices:
+            raise serializers.ValidationError("Feed type is not valid")
+        if attack_type not in attack_types:
+            raise serializers.ValidationError("Attack type is not valid")
+        if age not in age_choices:
+            raise serializers.ValidationError("Age is not valid")
+        if format_ not in formats:
+            raise serializers.ValidationError("Format is not valid")
```

Comment:
this is cool but don't you think we could simplify this further by leveraging ChoiceField? https://www.django-rest-framework.org/api-guide/fields/#choicefield

Reply:
oh thats a perfect fit I'll change it